### DISCUSSION TITLE
Unusually long build time, and huge amount of data consumed while attempting ./reach run

### DISCUSSION THREAD

**Discussion by Aro1914** [reach-sh/reach-lang#1230] [Q&A] [answered]
Hi, everyone. I finally ran my first reach DApp today!

But I'd like to share my experience before that actually happened.

Yesterday I successfully installed and setup my WSL 2 and Docker, then got my VSCode to have the necessary extensions. I started with the Rock, Paper, Scissors tutorial, after completing the setup for the first run, I ran 

```shell
$ ./reach run
```
I saw the build log, so I waited a little, but it turned out to be a very long wait. Eventually it failed,  so I tried again, and again. 
Then I noticed the traffic on my network was about 1.5MB/s so I decided to check my data balance and found out that I had used about 12GB running the build. I really don't know why it took so much, I reasoned that it may be because it was the first time running

```shell
$ ./reach run
```
But it was just too much
I had given up, but decided to try again today, since so much had been wasted already.
After a another long wait it occurred to me to take a screenshot
![Screenshot (372).png](https://user-images.githubusercontent.com/68448315/176906701-cd907f47-9da8-4460-9f09-f04126c41b1d.png)

As you can see it has been on for about 30 mins
I was really confused considering the amount of data spent, and the build had not yet completed

But after that try... I did another 

```shell
$ ./reach run
```
And it worked!
![Screenshot (376).png](https://user-images.githubusercontent.com/68448315/176907129-a6c50439-c93c-4666-8f83-7bcd697baa1a.png)

**Comment by jeapostrophe** [comment, marked as answer]
Wow! I'm sorry for this. I think you may have gotten help on Discord, but if not, let me add my two cents. This is really surprising to me! We went through a space optimization process with our images and you can see them here
- https://hub.docker.com/r/reachsh/runner/tags --- 180mb
- https://hub.docker.com/r/reachsh/reach/tags --- 72mb
- https://hub.docker.com/r/reachsh/reach-cli/tags --- 14mb
- https://hub.docker.com/r/reachsh/devnet-algo/tags --- 203mb

I have no idea how you got up 12gb. The only thing I can think of was that maybe the download kept failing and restarting and wasn't resuming from before? I'd like to try to figure out what that was, but I feel like I don't have the expertise to even really investigate.

**Reply by Aro1914** [reply]
Thanks for looking into the matter, but as you saw from the screenshot I provided indicating the long build, the downloads took a very long time, and ultimately failed to build. That screenshot was just one example, I had tried it several times the night before, and the results were the same. I'm new to Blockchain technology and so I thought that maybe because I was connected to the network that, that was the cause of the constant traffic, I never thought of the cause of the issue could be coming from my Docker image downloads because they never seemed to complete and none were above 100mb. I have been able to get my first build running now, but if there is anything that could conclusively be identified as the cause of the data consumed, I would be grateful to be notified about it.
